src/global_distrib/mod_global_distrib.erl

```diff
@@ -47,6 +48,129 @@ start(Host, Opts0) ->
+                                        validate = port}
+        },
+        process = fun ?MODULE:process_endpoints/1
+    }.
+
+tls_spec() ->
+    #section{
+        items = #{<<"enabled">> => #option{type = boolean},
+                  <<"certfile">> => #option{type = string,
+                                            validate = filename},
+                  <<"cacertfile">> => #option{type = string,
+                                              validate = filename},
+                  <<"ciphers">> => #option{type = string},
+                  <<"dhfile">> => #option{type = string,
+                                          validate = filename}
+        },
+        process = fun ?MODULE:process_tls/1
+    }.
+
+redis_spec() ->
+    #section{
+        items = #{<<"pool">> => #option{type = atom,
+                                        validate = pool_name},
+                  <<"expire_after">> => #option{type = integer,
+                                                validate = non_negative},
+                  <<"refresh_after">> => #option{type = integer,
+                                                 validate = non_negative}                              
+        }
+    }.
+
+cache_spec() ->
+    #section{
+        items = #{<<"cache_missed">> => #option{type = boolean},
+                  <<"domain_lifetime_seconds">> => #option{type = integer,
+                                                           validate = non_negative},
+                  <<"jid_lifetime_seconds">> => #option{type = integer,
+                                                        validate = non_negative},
+                  <<"max_jids">> => #option{type = integer,
+                                            validate = non_negative}
+        }
+    }.
+
+bounce_spec() ->
+    #section{
+        items = #{<<"enabled">> => #option{type = boolean},
+                  <<"resend_after_ms">> => #option{type = integer,
+                                                   validate = non_negative},
+                  <<"max_retries">> => #option{type = integer,
+                                               validate = non_negative}
+        },
+        process = fun ?MODULE:process_bounce/1
+    }.
+
+process_connections(KV) ->
+    proplists:substitute_aliases([{tls, tls_opts}], KV).
+
+process_endpoints([{host, Host}, {port, Port}]) ->
+    {Host, Port}.
+
+process_tls(KV) ->
+    case proplists:get_value(enabled, KV, false) of
+        true -> KV1 = proplists:delete(enabled, KV),
+                proplists:substitute_aliases([{cacertfile, cafile}], KV1);
+        false -> false
+    end.
+
+process_bounce([{enabled, false} | _]) ->
+    false;
+process_bounce([{enabled, true}, {max_retries, Retries}, {resend_after_ms, Resend}]) ->
+    [{max_retries, Retries}, {resend_after_ms, Resend}];
+process_bounce([{max_retries, Retries}, {resend_after_ms, Resend}]) ->
+    [{max_retries, Retries}, {resend_after_ms, Resend}].
```

Comment:
I am using `proplists:split` - it helps a lot, extracts multiple fields at a time and does not depend on ordering (which is undefined here):
```suggestion
process_bounce(KVs) ->
  {[EnabledOpts], Opts} = proplists:split(KVs),
  bounce_value(EnabledOpts, Opts).
  
bounce_value([{enabled, false}], _) -> false;
bounce_value(_, Opts) -> Opts.
```

Reply:
Yes I didn't refactor it properly as I got a bit frustrated before I realized that KV lists in function heads depend on ordering, in contrary to e.g. records.